src/components/routes/qdm/SDEPage.tsx

```diff
@@ -1,6 +1,205 @@
+  const { updateMeasure } = measureStore;
+  const { updateRouteHandlerState } = routeHandlerStore;
+  const [discardDialogOpen, setDiscardDialogOpen] = useState(false);
+  const [toastOpen, setToastOpen] = useState<boolean>(false);
+  const [toastMessage, setToastMessage] = useState<string>("");
+  const [toastType, setToastType] = useState<string>("danger");
+  const [currentSdeIncluded, setCurrentSdeIncluded] =
+    useState<boolean>(undefined);
+
+  useEffect(() => {
+    const subscription = measureStore.subscribe(setMeasure);
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, []);
+
+  useEffect(() => {
+    if (measure && measure.testConfiguration) {
+      setCurrentSdeIncluded(measure.testCaseConfiguration.sdeIncluded);
+    }
+  }, [measure]);
+
+  const formik = useFormik({
+    initialValues: {
+      sdeIncluded:
+        String(measure?.testCaseConfiguration?.sdeIncluded) || "true",
+    },
+    enableReinitialize: true,
+    onSubmit: async (values: SDEForm) => await handleSubmit(),
+  });
+  const { resetForm } = formik;
+
+  useEffect(() => {
+    updateRouteHandlerState({
+      canTravel: !formik.dirty,
+      pendingRoute: "",
+    });
+  }, [formik.dirty]);
+
+  const onToastClose = () => {
+    setToastType("danger");
+    setToastMessage("");
+    setToastOpen(false);
+  };
+  const handleToast = (type, message, open) => {
+    setToastType(type);
+    setToastMessage(message);
+    setToastOpen(open);
+  };
+
+  const onCancel = () => {
+    setDiscardDialogOpen(true);
+  };
+
+  const handleSubmit = async () => {
+    if (
+      currentSdeIncluded != undefined &&
+      formik.values.sdeIncluded !== String(currentSdeIncluded)
+    ) {
+      setCurrentSdeIncluded(formik.values.sdeIncluded === "true");
+      return;
+    }
+
+    const newMeasure: Measure = {
+      ...measure,
+      testCaseConfiguration: {
+        sdeIncluded: formik.values.sdeIncluded === "true",
+      },
+    };
+
+    measureServiceApi
+      .updateMeasure(newMeasure)
+      .then(() => {
+        handleToast(
+          "success",
+          "Test Case Configuration Updated Successfully",
+          true
+        );
+        // updating measure will propagate update state site wide.
+        updateMeasure(newMeasure);
+        setCurrentSdeIncluded(newMeasure.testCaseConfiguration.sdeIncluded);
+      })
+      // update to alert
+      .catch((err) => {
+        handleToast(
+          "danger",
+          "Error updating Test Case Configuration: " +
+            err?.response?.data?.message?.toString(),
+          true
+        );
+      });
+  };
+
+  return (
+    <form
+      id="sde-form"
+      onSubmit={handleSubmit}
+      data-testid={`sde-form`}
+      style={{ minHeight: 539 }}
+    >
+      <div className="content">
+        <div className="subTitle">
+          <h2>SDE</h2>
+        </div>
+        <div className="sde-page" data-testId="sde-page">
+          <RadioButton
```

Comment:
We may want to make sure who can make changes to this form.

Reply:
I added the canEdit tag